appinventor/components/src/com/google/appinventor/components/runtime/util/FileCache.java

```diff
@@ -0,0 +1,105 @@
+package com.google.appinventor.components.runtime.util;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FileCache {
+  public File cacheDir;
```

Comment:
I don't know why this needs to be public. If it needs to be accessed from somewhere, make a getter function instead.